java/client/src/test/java/glide/connection/ConnectionWithGlideMockTests.java

```diff
@@ -0,0 +1,206 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.connection;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import connection_request.ConnectionRequestOuterClass.ConnectionRequest;
+import connection_request.ConnectionRequestOuterClass.NodeAddress;
+import glide.api.RedisClient;
+import glide.api.models.exceptions.ClosingException;
+import glide.connectors.handlers.CallbackDispatcher;
+import glide.connectors.handlers.ChannelHandler;
+import glide.connectors.resources.Platform;
+import glide.managers.CommandManager;
+import glide.managers.ConnectionManager;
+import glide.utils.RustCoreLibMockTestBase;
+import glide.utils.RustCoreMock;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import redis_request.RedisRequestOuterClass.RedisRequest;
+import response.ResponseOuterClass.Response;
+
+public class ConnectionWithGlideMockTests extends RustCoreLibMockTestBase {
```

Comment:
Can we add tests for InterruptedExceptions on the thread?

Reply:
What should I do to get one? Kill a thread?

Reply:
yes?